src/mender/scripts/aggregator/inventory.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import logging as log
+import os
+import os.path as path
+import subprocess
+
+from mender.scripts.aggregator.aggregator import ScriptKeyValueAggregator
+import mender.scripts.artifactinfo as artifactinfo
+import mender.scripts.devicetype as devicetype
+
+
+def aggregate(
+    script_path="",
+    device_type_path="",
+    artifact_info_path="",
+):
+    """Runs all the inventory scripts in 'path', and parses the 'key=value' pairs
+    into a data-structure ready for passing it on to the Mender server"""
+    log.info("Aggregating inventory data from {script_path}")
+    keyvals = {}
+    for inventory_script in inventory_scripts(script_path):
+        keyvals.update(inventory_script.run())
+    dt = devicetype.get(device_type_path)
+    log.info(f"Found the device type: {dt}")
+    if dt:
+        keyvals.update(dt)
+    an = artifactinfo.get(artifact_info_path)
+    log.info(f"Found the artifact_name: {an}")
+    if an:
+        keyvals.update(an)
+    return keyvals
+
+
+def inventory_scripts(inventory_dir):
```

Comment:
We miss the type annotations through all the code. Given the fact we are starting a new project, I would suggest to take advantage of the Python type annotations and to include mypy in our QA pipeline. We won't regret this choice, as having `mypy` in since day zero is way easier than introducing it afterwards.

For example, here you would write:
```suggestion
from typing import List

def inventory_scripts(inventory_dir: str) -> List[ScriptKeyValueAggregator]:
```